release-notes/6.0/api-diff/.Net/6.0.0_System.IO.md

```diff
@@ -0,0 +1,112 @@
+# System.IO
+
+``` diff
+ namespace System.IO {
+     public class BinaryReader : IDisposable {
++        public virtual Half ReadHalf();
+     }
+     public class BinaryWriter : IAsyncDisposable, IDisposable {
++        public virtual void Write(Half value);
+     }
+     public sealed class BufferedStream : Stream {
+-        public override int Read(byte[] array, int offset, int count);
++        public override int Read(byte[] buffer, int offset, int count);
+-        public override void Write(byte[] array, int offset, int count);
++        public override void Write(byte[] buffer, int offset, int count);
+     }
+     public static class Directory {
++        public static FileSystemInfo CreateSymbolicLink(string path, string pathToTarget);
++        public static FileSystemInfo ResolveLinkTarget(string linkPath, bool returnFinalTarget);
+     }
+     public class EnumerationOptions {
++        public int MaxRecursionDepth { get; set; }
+     }
+     public static class File {
++        public static FileSystemInfo CreateSymbolicLink(string path, string pathToTarget);
++        public static FileStream Open(string path, FileStreamOptions options);
++        public static SafeFileHandle OpenHandle(string path, FileMode mode = FileMode.Open, FileAccess access = FileAccess.Read, FileShare share = FileShare.Read, FileOptions options = FileOptions.None, long preallocationSize = (long)0);
++        public static FileSystemInfo ResolveLinkTarget(string linkPath, bool returnFinalTarget);
```

Comment:
With this, cant we have it where File.ReadAllBytes/Text (sync and async), and File.WriteAllBytes/Text (sync and async) check if they are trying to read from/write to a symbolic link so that way they do not unintetionally overwrite the link with the actual file contents when they intend it to actually write to the file located where the symbolic link points to?

Or was that already done for .NET 6? I was bitten by that issue before in one of my projects 😭.